x/mongo/driver/session/client_session.go

```diff
@@ -79,6 +81,30 @@ func (s TransactionState) String() string {
 	}
 }
 
+// LoadBalancedTransactionConnection represents a connection that's pinned by a ClientSession because it's being used
+// to execute a transaction when running against a load balancer. This interface is a copy of driver.PinnedConnection
+// and exists to be able to pin transactions to a connection without causing an import cycle.
```

Comment:
The copy seems good to me, although I find the name a little confusing given it's a copy of `driver.PinnedConnection`. Why not something like `session.PinnedTransactionConnection`?

The compile-time assertion seems like it'd be fine [here](https://github.com/mongodb/mongo-go-driver/blob/master/x/mongo/driver/driver.go#L75) with a comment.

Reply:
The idea was that the name of the copy should clearly indicate its purpose, which is to function as a pinned connection for load-balanced transactions, so I'd prefer to keep it. We can talk about changing it if you feel strongly, though.

Added the assertion and comment.

Reply:
Fine with me, just thought I'd bring it up.

Assertion looks good; thanks for documenting everything so thoroughly.